### PR TITLE
DOCSP-17867 node 4.1 release

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info

Updated the:
- compatibility table to include 4.1
- what's new page to have the new features of 4.1 mentioned here[
](https://github.com/mongodb/node-mongodb-native/releases/tag/v4.1.0)

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-17867

### Docs staging link (requires sign-in on MongoDB Corp SSO):

https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-17867-Node4.1Release/compatibility/

https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-17867-Node4.1Release/whats-new/#what-s-new-in-4.1

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -12,9 +12,20 @@ What's New
...
 * :ref:`Version 3.6 <version-3.6>`
 
+.. _version-4.1:
+
+What's New in 4.1
+-----------------
+
+New features of the 4.1 Node.js driver release include:
+
+- Added support for load balancer to specify how a driver should behave when connected
```
Comment (terakilobyte) [inline source/whats-new.txt:26]: What does this provide the user? And is there a link we can provide after expanding this?

File: source/whats-new.txt [hunk 2]
```diff
@@ -12,9 +12,20 @@ What's New
...
+- Added support for the ``advanceClusterTime()`` method to update the cluster time tracked by a session
```
Comment (terakilobyte) [inline source/whats-new.txt:27]: Same as the previous comment.

**Review by terakilobyte — CHANGES_REQUESTED** [review]
I think we should provide more information about these new features.

File: source/whats-new.txt (hunk 1 above)
Comment (kyuan-mongodb) [reply source/whats-new.txt]: I couldn't find any existing docs on load balancer, so I'm thinking to make that part of the ticket to document it

**Review by terakilobyte — APPROVED** [review]
lgtm

**Comment by kyuan-mongodb** [comment]
Hi @emadum and @nbbeeken, Can I get a review to make sure everything in this PR is correct?

**Comment by kyuan-mongodb** [comment]
@mongodb/dbx-node

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**